Plugins/IngameDebugConsole/Scripts/DebugLogConsole.cs

```diff
@@ -197,18 +197,29 @@ public static void SearchAssemblyForConsoleMethods( Assembly assembly )
 		{
 			try
 			{
-				foreach( Type type in assembly.GetExportedTypes() )
+				Type[] types = assembly.GetExportedTypes();
+				List<ConsoleAttribute> methods = new();
+				const BindingFlags BindingAttr = BindingFlags.Static | BindingFlags.Public | BindingFlags.DeclaredOnly;
+				for (int t = 0; t < types.Length; t++)
 				{
-					foreach( MethodInfo method in type.GetMethods( BindingFlags.Static | BindingFlags.Public | BindingFlags.DeclaredOnly ) )
+					MethodInfo[] typeMethods = types[t].GetMethods(BindingAttr);
+					for (int m = 0; m < typeMethods.Length; m++)
 					{
-						foreach( object attribute in method.GetCustomAttributes( typeof( ConsoleMethodAttribute ), false ) )
+						MethodInfo method = typeMethods[m];
+						IEnumerable<ConsoleAttribute> attributes = method.GetCustomAttributes<ConsoleAttribute>(true);
+						foreach (ConsoleAttribute attribute in attributes)
 						{
-							ConsoleMethodAttribute consoleMethod = attribute as ConsoleMethodAttribute;
-							if( consoleMethod != null )
-								AddCommand( consoleMethod.Command, consoleMethod.Description, method, null, consoleMethod.ParameterNames );
+							attribute.SetMethod(method);
+							methods.Add(attribute);
```

Comment:
Thank you for the changes.

- Could we make the diff minimalistic by changing only the necessary lines? This will help with Git's Blame/Annotate functionality. Ideally, only the code inside the innermost foreach should change.
- I'd advise against casting arrays or lists to IEnumerable because `foreach(list)` and `foreach(array)` (these use duck typing) are more optimized than `foreach(ienumerable)` (you can check out their GC allocations in Profiler).
- new() isn't supported on older Unity versions. I also prefer this approach in my personal projects but atm not in my plugins.

Reply:
>     * Could we make the diff minimalistic by changing only the necessary lines? This will help with Git's Blame/Annotate functionality. Ideally, only the code inside the innermost foreach should change.

Sure, I personally try to always for loops over foreach loops. 

>     * I'd advise against casting arrays or lists to IEnumerable because `foreach(list)` and `foreach(array)` (these use duck typing) are more optimized than `foreach(ienumerable)` (you can check out their GC allocations in Profiler).

GetCustomAttributes returns a IEnumerable and not a list or array.

>     * new() isn't supported on older Unity versions. I also prefer this approach in my personal projects but atm not in my plugins.

Force of habit, will fix!